### ISSUE TITLE
Update embed_sub to populate all data extensions

### ISSUE THREAD

**Issue by sosey** [spacetelescope/wfc3tools#91]
see also https://github.com/spacetelescope/slitlessutils/issues/81 for the full explanation

The current form of these programs is to only embed subarrays onto the extensions relevant to the detector they belong to, but slitlessutils is expecting to loop over all detectors even if they are null.

@rosteen @larrybradley @bjkuhn

**Comment by sosey** [comment]
@pllim @rosteen does the same subarray embed routine exist in ACS for specifcally working with spectral data, I didn't see it offhand, but we should make sure it's inline with what slitlessutils is expecting, and create a ticket over there too.

**Comment by pllim** [comment]
I am not aware of any such functionality over at acstools (https://acstools.readthedocs.io/en/latest/). I also don't have ACS WBS anymore so I am going to ping @jryon and @mdlpstsci here, just in case they know of such thing elsewhere. 🙏

**Comment by mdlpstsci** [comment]
I am not aware of this type of functionality in the ACSTOOLS.

**Comment by jryon** [comment]
acstools doesn't currently have functionality for this, though I'm working on it to enable serial CTE correction of subarrays. (Serial CTE correction in CALACS can only operate on full-frames.) I have some basic code that can embed a subarray into a full-frame it for ACS/WFC, but does not mess with WCS keywords, which I'd guess are needed for slitless-related things? I'm also not sure how general it is, as I've only tested it for standard ACS/WFC subarrays.

**Comment by rosteen** [comment]
I wouldn't close this yet, but FYI it seems like it makes more sense to properly handle the subarray case downstream in `slitlessutils` rather than forcing `wfc3tools` to add in empty extensions and such. I'm taking a crack at that to confirm it's not too bad before closing this issue.

**Comment by pllim** [comment]
@rosteen , was this ever addressed in slitlessutils?

**Comment by rosteen** [comment]
Yes, this was addressed via https://github.com/spacetelescope/slitlessutils/pull/110 and can be closed.
